### PR TITLE
More aggressively market functorch.vmap when torch.vmap gets called

### PR DESCRIPTION
**Pull request by zou3519**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#67347 More aggressively market functorch.vmap when torch.vmap gets called**

This PR:
- changes the warning when torch.vmap gets called to suggest using
functorch.vmap
- changes the warning when a batching rule isn't implemented to suggest
using functorch.vmap

Test Plan:
- test/test_vmap.py

Differential Revision: [D31966603](https://our.internmc.facebook.com/intern/diff/D31966603)

### CODE REVIEW

File: aten/src/ATen/BatchedFallback.cpp [hunk 1]
```diff
@@ -66,13 +66,16 @@ static bool isInplaceOp(const c10::FunctionSchema& schema) {
...
   if (!globalContext().areVmapFallbackWarningsEnabled()) {
     return;
   }
-  auto uses_stack = is_inplace ? "" : " and stack";
-  TORCH_WARN("Batching rule not implemented for ", schema.operator_name(), " falling back "
-             "to slow (for loop", uses_stack, ") implementation");
+  TORCH_WARN("There is a performance drop because we have not yet implemented ",
+             "the batching rule for ", schema.operator_name(), ". ",
+             "We've moved development of vmap to to functorch "
+             "(https://github.com/pytorch/functorch), please try functorch.vmap) "
```
Comment (samdow) [inline aten/src/ATen/BatchedFallback.cpp:76]: nit: Unmatched paren after `functorch.vmap`

File: aten/src/ATen/BatchedFallback.cpp [hunk 2]
```diff
@@ -66,13 +66,16 @@ static bool isInplaceOp(const c10::FunctionSchema& schema) {
...
+             "instead and/or file ",
+             "us an issue on GitHub so that we can prioritize its implementation.");
```
Comment (samdow) [inline aten/src/ATen/BatchedFallback.cpp:78]: nit: "file us an issue" -> "file an issue"

File: torch/_vmap_internals.py [hunk 3]
```diff
@@ -158,9 +158,10 @@ def vmap(func: Callable, in_dims: in_dims_t = 0, out_dims: out_dims_t = 0) -> Ca
...
-        We are actively developing a different and improved vmap prototype
-        `here. <https://github.com/zou3519/functorch>`_ The improved
-        prototype is able to arbitrarily compose with gradient computation.
+        We have moved development of vmap to
+        `functorch. <https://github.com/pytorch/functorch>`_ functorch's
+        vmap is able to arbitrarily compose with gradient computation
+        and contains significant performance improvements.
         Please give that a try if that is what you're looking for.
 
         Furthermore, if you're interested in using vmap for your use case,
```
Comment (samdow) [inline torch/_vmap_internals.py:167]: Do we want to update this to send people to functorch instead so things don't get lost in a pytorch issue?

**Review by samdow — APPROVED** [review]
LGTM!

File: torch/_vmap_internals.py (hunk 3 above)
Comment (zou3519) [reply torch/_vmap_internals.py:167]: That pytorch issue directs folks back to functorch (and also has instructions for how to find us on Slack) so I think it's fine to leave it for now.

**Comment by zou3519** [comment]
@zou3519 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31966603).

**Pull request closed without merging**